jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -215,6 +224,42 @@ public Entry acquire()
         return null;
     }
 
+    /**
+     * Utility method to acquire an entry from the pool,
+     * reserving and creating a new entry if necessary.
+     *
+     * @param creator a function to create the pooled value for a reserved entry.
+     * @return an entry from the pool or null if none is available.
+     */
+    public Entry acquire(Function<Entry, T> creator)
+    {
+        Entry entry = acquire();
+        if (entry != null)
+            return entry;
+
+        Reservation reservation = reserve(getMaxEntries());
+        if (reservation == null)
+            return null;
+
+        T value = null;
+        try
+        {
+            value = creator.apply(reservation.getEntry());
+        }
+        catch (Throwable th)
+        {
+            LOGGER.warn(th);
```

Comment:
I think we should call `reservation.remove()` and rethrow here - same semantic of `Map.computeIfAbsent()`.

Reply:
just throw and not warn?   I don't really like log and throw.